src/server/blockchain.hpp

```diff
@@ -33,6 +33,7 @@ class BlockChain {
         uint32_t findBlockForTransaction(Transaction &t);
         uint32_t findBlockForTransactionId(SHA256Hash txid);
         ExecutionStatus addBlock(Block& block);
+        ExecutionStatus addBlockSync(Block& block);
```

Comment:
Since the lock is moved out of addBlock it should be made private method now so it won't be accidentally used outside Blockchain in future.